importer/tag/src/main/java/org/itsallcode/openfasttrace/importer/tag/LongTagImportingLineConsumer.java

```diff
@@ -53,10 +56,12 @@ public void processMatch(final Matcher matcher, final int lineNumber, final int
     {
         this.listener.beginSpecificationItem();
         this.listener.setLocation(this.file.getPath(), lineNumber);
-        final SpecificationItemId coveredId = SpecificationItemId.parseId(matcher.group(2));
-        final List<String> neededArtifactTypes = parseCommaSeparatedList(matcher.group(6));
-        final SpecificationItemId generatedId = generateItemId(lineNumber, lineMatchCount, coveredId, matcher.group(1),
-                neededArtifactTypes);
+        final SpecificationItemId coveredId = SpecificationItemId.parseId(matcher.group(3));
+        final List<String> neededArtifactTypes = parseCommaSeparatedList(matcher.group(7));
+        final String artifactType = matcher.group(1);
+        final Optional<String> revision = Optional.ofNullable(matcher.group(2));
+        final SpecificationItemId generatedId = generateItemId(lineNumber, lineMatchCount, coveredId, artifactType,
```

Comment:
Passing `Optional` as method parameter considered harmful:
https://www.baeldung.com/java-optional#misuages

Money quote:
> The intent of Java when releasing Optional was to use it as a return type

There is a reason why I am not a fan o Optional. It does not remove the potential for the parameter being null for example. In essence, Optional should neither be used as parameter nor as class attribute. 

A cleaner way would be overloading methods. Even a `null` check is in the end more elegant.